Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -533,6 +534,16 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
                 m_readyState = MediaPlayer::ReadyState::HaveMetadata;
                 // FIXME: Should we manage NetworkState too?
                 GST_DEBUG("m_readyState=%s", dumpReadyState(m_readyState));
+            } else if (mseBuffering) {
+                // The HAVE_CURRENT_DATA indicates that the data for immediate current playback position
+                // is in place but the playback might not be able to move forward smoothly. Buffering in non-mse
+                // sense is something like this because it triggers when there's still some data in the buffer
+                // but a low watermark was hit. However in MSE case when the data for the current position is not
+                // there in the buffer it's rather HAVE_META_DATA which means "No media data is available for
+                // the immediate current playback position."
+                m_readyState = MediaPlayer::ReadyState::HaveMetadata;
```

Comment:
Would it make more sense to use a more fine grained readyState management for the general case, like the one [introduced in this commit (focused on seeks)](https://github.com/WebPlatformForEmbedded/WPEWebKit/commit/6caf899f8a7ab94bae2f288b464b475ce6175d0e)?

Reply:
After discussing the topic with you in private, it looks like a finer grained readyState management would interfere with the duties of SourceBuffer/MediaSource managing the readyState, so it's better to only correct it here in extreme cases (HaveMetadata when we're completely sure that the playback can't go on). This change is fine as originally proposed.